clients/src/main/resources/common/message/ConsumerGroupHeartbeatResponse.json

```diff
@@ -0,0 +1,70 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "apiKey": 68,
+  "type": "response",
+  "name": "ConsumerGroupHeartbeatResponse",
+  "validVersions": "0",
+  "flexibleVersions": "0+",
+  // Supported errors:
+  // - GROUP_AUTHORIZATION_FAILED
+  // - NOT_COORDINATOR
+  // - COORDINATOR_NOT_AVAILABLE
+  // - COORDINATOR_LOAD_IN_PROGRESS
+  // - INVALID_REQUEST
+  // - UNKNOWN_MEMBER_ID
+  // - FENCED_MEMBER_EPOCH
+  // - UNSUPPORTED_ASSIGNOR
+  // - UNRELEASED_INSTANCE_ID
+  // - GROUP_MAX_SIZE_REACHED
+  "fields": [
+    { "name": "ThrottleTimeMs", "type": "int32", "versions": "0+",
+      "about": "The duration in milliseconds for which the request was throttled due to a quota violation, or zero if the request did not violate any quota." },
+    { "name": "ErrorCode", "type": "int16", "versions": "0+",
+      "about": "The top-level error code, or 0 if there was no error" },
+    { "name": "ErrorMessage", "type": "string", "versions": "0+", "nullableVersions": "0+", "default": "null",
+      "about": "The top-level error message, or null if there was no error." },
+    { "name": "MemberId", "type": "string", "versions": "0+", "nullableVersions": "0+", "default": "null",
+      "about": "The member id generated by the coordinator. Only provided when the member joins with MemberEpoch == 0." },
+    { "name": "MemberEpoch", "type": "int32", "versions": "0+",
+      "about": "The member epoch." },
+    { "name": "ShouldComputeAssignment", "type": "bool", "versions": "0+",
+      "about": "True if the member should compute the assignment for the group." },
+    { "name": "HeartbeatIntervalMs", "type": "int32", "versions": "0+",
+      "about": "The heartbeat interval in milliseconds." },
+    { "name": "Assignment", "type": "Assignment", "versions": "0+", "nullableVersions": "0+", "default": "null",
+      "about": "null if not provided; the assignment otherwise.", "fields": [
+      { "name": "Error", "type": "int8", "versions": "0+",
+        "about": "The assigned error." },
+      { "name": "AssignedTopicPartitions", "type": "[]TopicPartitions", "versions": "0+",
+        "about": "The partitions assigned to the member that can be used immediately." },
+      { "name": "PendingTopicPartitions", "type": "[]TopicPartitions", "versions": "0+",
+        "about": "The partitions assigned to the member that cannot be used because they are not released by their former owners yet." },
+      { "name": "MetadataVersion", "type": "int16", "versions": "0+",
+        "about": "The version of the metadata." },
+      { "name": "MetadataBytes", "type": "bytes", "versions": "0+",
+        "about": "The assigned metadata." }
+    ]}
+  ],
+  "commonStructs": [
```

Comment:
what's the difference between fields and commonStructs?

Reply:
Common structs can be used multiple times when writing the json. It prevents repeated text in the fields.

Reply:
See TopicPartitions used in line 52, 54. If we didn't have the common struct, we'd have to define the name and the nested fields each time.